src/components/anonymous-modal.jsx

```diff
@@ -1,45 +1,42 @@
-import React from 'react';
 import * as BS from 'react-bootstrap';
 import {GearIcon} from 'react-octicons';
 
 import CurrentUserStore from '../user-store';
 
 let hasAlreadyShownAnonymousModal = false;
 
-const AnonymousModal = React.createClass({
-  render() {
-    const onHide = () => {
-      hasAlreadyShownAnonymousModal = true;
-      this.setState({ showModal: false});
-    };
-    let showModal;
-    if (CurrentUserStore.getUser()) {
-      showModal = false;
-    } else {
-      showModal = !hasAlreadyShownAnonymousModal;
-    }
+function AnonymousModal() {
+  const onHide = () => {
+    hasAlreadyShownAnonymousModal = true;
+    this.setState({ showModal: false});
```

Comment:
This will introduce error, since it's functional component and has no this.setState
https://jsbin.com/dodalubaxo/edit?html,js,output

Reply:
And the worst part is that this modal is showed when you visit the issues view
https://deploy-preview-17--coala-gh-board.netlify.com/#/r/coala:coala%7Ccoala-bears%7Cdocumentation%7Ccorobo/kanban